File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.py
from collections import Counter

def solution(X, Y):
    counter_X = Counter(X)
    counter_Y = Counter(Y)

    common = []
    for digit in map(str, range(10)):
        count = min(counter_X[digit], counter_Y[digit])
        common.extend([digit] * count)

    if not common:
        return "-1"

    common.sort(reverse=True)
    result = ''.join(common)

    if result[0] == '0':  # 모든 숫자가 0이면
        return "0"

    return result

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.js
function solution(X, Y) {
    let countX = {}, countY = {};
    [...X].forEach(x => countX[x] = (countX[x] || 0) + 1);
    [...Y].forEach(y => countY[y] = (countY[y] || 0) + 1);

    let common = [];
    for(let i = 0 ; i <= 9 ; i++){
        const digit = i.toString();
        const digitCnt = Math.min(countX[digit] || 0, countY[digit] || 0);
        
        for(let j = 0 ; j < digitCnt ; j++){
            common.push(digit);
        }
    }
    
    if(common.length === 0) return "-1";
    
    const result = common.sort((a, b) => b - a).join("");
    return result[0] === "0" ? "0" : result;
}